state/machine_linklayerdevices.go

```diff
@@ -489,15 +427,6 @@ func (m *Machine) setDevicesFromDocsOps(newDocs []linkLayerDeviceDoc) ([]txn.Op,
 
 func (m *Machine) insertLinkLayerDeviceOps(newDoc *linkLayerDeviceDoc) ([]txn.Op, error) {
 	var ops []txn.Op
-	if newDoc.ParentName != "" {
-		newParentDocID, err := m.parentDocIDFromDeviceDoc(newDoc)
-		if err != nil {
-			return nil, errors.Trace(err)
-		}
-		if newParentDocID != "" {
-			ops = append(ops, assertLinkLayerDeviceExistsOp(newParentDocID))
```

Comment:
Does the facade code guarantee that devices are inserted in the correct order (and/or uses such an assertion in the generated txn op list)? If not, how do we ensure that we are not missing any parents?

Reply:
The facade code does not call though this path, but you have turned up that we do not actually have this check.

We *do* note observed parent devices up-front in order to ensure that devices referred to by children are not removed. I'll add another ticket for this part.